solutions/usaco-989.mdx

```diff
@@ -2,46 +2,108 @@
 id: usaco-989
 source: USACO Bronze 2020 January
 title: Race
-author: Sathvik Chundru
+author: Sathvik Chundru, Kevin Sheng
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_race_bronze_jan20.html)
 
 <LanguageSection>
- 
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+
+using std::cout;
+using std::endl;
+
+int fastest_time(int dist, int max_speed) {
+	int lhs_travel = 0;  // Amount of distance where Bessie's speeding up
+	int rhs_travel = 0;  // and slowing down respectively
+	int time = 0;
+	
+	// Gradually speed up until we achieve our distance
+	for (int curr_speed = 1; ; curr_speed++) {
+		lhs_travel += curr_speed;
+		time++;
+		if (lhs_travel + rhs_travel >= dist) { 
+			return time; 
+		}
+
+		/*
+		 * If we're above the speed limit, add the current speed
+		 * to the part where we slow down (the RHS) as well.
+		 */
+		if (curr_speed >= max_speed) {
+			rhs_travel += curr_speed;
+			time++;
+			// Check again if we've achieved our distance
+			if (lhs_travel + rhs_travel >= dist) { 
+				return time; 
+			}
+		}
+	}
+}
+
+int main() {
+	std::ifstream read("race.in");
+	int race_len;
+	int query_num;
+	read >> race_len >> query_num;
+
+	std::ofstream written("race.out");
```

Comment:
```suggestion
	std::ofstream write("race.out");
```

Reply:
why

Reply:
consistent verb tense smh

Reply:
past tense of read is read

Reply:
why are you even using past tense?

Reply:
"this is the file that is read/written"
at this point it's personal preference, just like `t` vs. `test_num`

Reply:
you "read" in this file and "write" to this file.